docs/pipenv-developer-usage.md

```diff
@@ -0,0 +1,72 @@
+# Pipenv developers' guide
```

Comment:
merge in this document also the information available at https://github.com/Alfresco/alfresco-ansible-deployment/blob/master/README.md#development and make this a 'developer guide' (not only pipenv-focused)

Reply:
Added this section and changed file title as well